### PR TITLE
Community Page Updates + New Meetups Page

### PR DESCRIPTION
**Pull request by sandraor**
I need some help to finish the new Meetups page. It's using Google Maps for Rails: https://github.com/apneadiving/Google-Maps-for-Rails

### Code
- [x] Show all groups in the meetup list. The meetups list has 5 groups (pulled from meetups.yml). It is set up to show 3 columns with 1 group per column, so the remaining 2 groups are currently hidden. Can we still have 3 columns but include all of the groups (for example, show Europe directly under South America)? (@andrewbranch)
- [x] Pull marker and info window information from meetups.yml: Latitude, longitude, location, organizer name (can be multiple), organizer image (can be multiple), url. (@lzcabrera & @solsberg)
- [x] Link list items with their markers. See "Sidebar" example on http://apneadiving.github.io/. Add an "active" class to the list items when they are active (@lzcabrera)
- [x] Initially set the map to the user's location. If location detection is disabled, just show the map as usual. See https://developers.google.com/maps/documentation/javascript/examples/map-geolocation (@wifelette)
- [x] Customize marker cluster icons. I just need a place to add a URL and customize the font colour. See https://github.com/apneadiving/Google-Maps-for-Rails/wiki/Customized-Cluster-Icons (@alexclarkofficial)
- [x] Clean up the current javascript. I'm not sure if was put in the right places. (@yaymukund)
- [x] When relevant, have organizer data pulled in directly from Meetup.com (@lzcabrera) 

### Content
- [x] Fill in any missing information (the "Start your own meetup" section in particular). (@wifelette)

Names in parenthesis indicate the tasks have been claimed. Then we'll check them off once the work is complete and merged back in.

### CODE REVIEW

**Comment by wifelette** [comment]
This is awesome :D 

I'll work on the missing info section content.

Hoping we can get some help here from some others...

**Comment by wycats** [comment]
@sandraor Do you think you could rebase this PR? It seems to have diverged somewhat from master, and I'm not personally sure how to resolve all of the conflicts.

**Comment by sandraor** [comment]
@wycats Sure — it's up to date now.

File: data/meetups.yml [hunk 1]
```diff
@@ -102,19 +101,23 @@
...
     url:      http://www.meetup.com/Amsterdam-Ember-js/
+"Asia":
+  - location: Tel Aviv, Israel
+    url:      http://www.meetup.com/Ember-IL/
+    image:    20telaviv.png
   - location: Bangalore, India
     url:      http://www.meetup.com/EmberJS-Bangalore/
   - location: Tokyo, Japan
     url:      http://emberjs.doorkeeper.jp/
+"Australia":
```
Comment (wifelette) [inline data/meetups.yml]: How about South Pacific?

Comment (martinstannard) [inline data/meetups.yml]: [Oceania](http://en.wikipedia.org/wiki/Oceania) seems the correct term for the region.

**Comment by andrewbranch** [comment]
Is anyone already working on the meetups group columns? Thinking about trying to tackle that. Would a Middleman extension be the best place to do some logic to try to balance the groups into even columns?

**Comment by wifelette** [comment]
@andrewbranch I can't comment on the Middleman bits... but the task is all yours, thanks for the help :D

**Comment by wifelette** [comment]
@sandraor OKAY! It's way beyond my bedtime, but I got excited about making progress on this :p 

My latest PR tackles several of the ToDos.

Right now, the map is doing the geolocation (assuming you give the browser permission to detect), but there's only one marker on it, Atlanta. Looks like you hardcoded that in, yes?

Question being, what's the plan for getting them all in there? Do we need to add lat/long parameters to the yaml file? Same question RE: organizer info—looks like you hardcoded it for now. What's the plan now that we're ready to add everyone else's data?

Also, something strange that I was only able to repro a couple times, so I'm unsure if it's still a problem right now. The Ember logo (in the nav) was jumping up in size ever so slightly only on this new Meetup page... let me know if you see it. If not, since I was only able to repro intermittently, we can drop until we see it again :p

**Comment by sandraor** [comment]
@wifelette Correct. The hardcoded meetups data (lat, long, organizer info) needs to be added to the .yml file and pulled into meetups.js.

I haven't seen the logo change size yet. I'll watch for it!

**Comment by lzcabrera** [comment]
I'm working on the meetup data (lat, long, organizer info) and it's parsing to generate the map markers in meetup.js from the data on the .yml file.

**Comment by wifelette** [comment]
@lzcabrera Awesome! 

For both you and @andrewbranch: let us know if Sandra or I can help answer any questions to keep things moving. And thoughts on ETAs? I don't have a sense of how long these two tasks might take.

**Comment by wifelette** [comment]
I take it back. I forgot I was waiting on a review of @andrewbranch's branch. I'll point my nagging in that direction ;)

**Comment by lzcabrera** [comment]
I pulled in pull request #6 and i'm seeing errors on meetups.html.erb due to the updates I added to the meetups.yml structure. @andrewbranch any chance you can help me integrate the column balancer code with the meetups.yml file from pull request #7?

I updated the meetups.yml file locations to look something like this (notice the newly added area and groups attributes under locations):

locations:
  - area: North America
    groups:
      - location: San Francisco, CA
        url:      http://www.meetup.com/Ember-SF/
        image:    07sanfrancisco.png
     - location: New York, NY
        url:      http://www.meetup.com/EmberJS-NYC/
        image:    04newyork.png
  - area: South America
    groups:
      - location: São Paulo, Brazil
        url:      http://www.meetup.com/Ember-js-Sao-Paulo/
      - location: Belo Horizonte, Brazil
        url:      http://www.meetup.com/Ember-BH/
      - location: Montevideo, Uruguay
        url:      http://www.meetup.com/ember-montevideo/

**Comment by andrewbranch** [comment]
@lzcabrera Done! (lzcabrera/website#1)

**Comment by lzcabrera** [comment]
@andrewbranch thanks!

**Comment by lzcabrera** [comment]
@solsberg I'm done with task 2 but l'm with @wifelette anything that involve less manual data entry is always welcomed :)

Also, I'm seeing the following error on my console: http://localhost:4567/images/meetups/map-pin_x2.png Failed to load resource: the server responded with a status of 404 (Not Found)

Where can I find the vector graphic to generate the retina friendly version of the map-pin?

**Comment by sandraor** [comment]
Thanks @lzcabrera and @andrewbranch! I just merged your work.

I can take care of the retina version of map-pin.png since I have the vector file :) Thank you for noticing, @lzcabrera.

**Comment by wifelette** [comment]
Once we get https://github.com/sandraor/website/pull/3 merged in, I think the only thing left is "Customize marker cluster icons", yes? Or did someone grab that already and I missed it...

**Comment by alexclarkofficial** [comment]
Hey,
I was trying to take a stab at "Customize marker cluster icons" and I think I fixed a couple bugs with checking if `navigator.geolocation` is available and how `buildMap` is called.  Should I send a PR?

**Comment by wycats** [comment]
@alexclarkofficial Absolutely! I helped @wifelette get it started but it definitely was the bare minimum, and the timing of the centering wasn't quite right.

PR plz!

**Comment by alexclarkofficial** [comment]
Ok, awesome! I'm a little unsure of where to send it though on a multistep PR. Should I PR to the main project or @sandraor's branch?

**Comment by rwjblue** [comment]
@alexclarkofficial - @sandraor's branch

**Comment by alexclarkofficial** [comment]
@rwjblue Thanks! Think I messed it up for a minute there, but I think I got it now.

**Comment by jgwhite** [comment]
This is looking really really awesome :D

Any loose ends I can help out with?

**Comment by wifelette** [comment]
@lzcabrera I know you added a bunch of the data on the group organizers... question: did you investigate the [Meetups.com API](http://www.meetup.com/meetup_api/)? Not all the groups use that site, but most do, so I wonder if there's an easy way to set a flag so our code knows it's Meetup, and then auto-pulls data from there.

Might be too complex to be worth it, but just wondering if you investigated.

Mostly unrelated: @sandraor Do we need CSS to correct for organizer images that are the wrong size? On things like the logos, we just manually make sure things are the right size before merging, which is of course still an option here... but if we *can* correct for it in the code, even better. I was just noticing when poking through the code that the images there now (small headshots grabbed from Meetup sites I presume) aren't all the same size.

@jgwhite Nothing obvious right this sec, but stick around ;) Oh and by self-admission, most of us working on the various bits of this particular PR are newbs, either to GitHub or to JS. Feel free to poke around and submit or suggest tweaks/refactors.

**Comment by jgwhite** [comment]
You’ve all done a splendid job so far. If there are any chores or drudge work needs doing… I’ll gladly take it off your hands.

**Comment by lzcabrera** [comment]
I was looking into the Meetups.com API but then I noticed that the URL to some meetups were not pointing to meetup.com (London, Kyiv, Tokyo). Kyiv, Tokyo and Goteberg don't have a meetup page.

I think we now know the edge cases and we can plan around them.

I was trying to be agile and figured the meetup API could be hooked up later :). Also I was a bit unsure on how to handle the API Key so that it is not public. I guess I should have asked. 

I can look into integrating the meetup API. If I'm not mistaken I believe @solsberg is looking into creating a json file in @sandraor/website#3. So we should wait for him to be done with that so that the meetup data goes in the same json file with the map coordinates.

**Comment by wifelette** [comment]
All sounds sane :smile: 

In case it's not clear, I'm not at all *certain* about what the Meetup API is capable of and whether or not it'll be useful—I just wanted us to investigate. And yep, getting this out the door and continuing to investigate that bit makes perfect sense.

**Comment by solsberg** [comment]
OK. I have added a [new pull request](https://github.com/sandraor/website/pull/10) based on a new branch which is intended to replace [my earlier one](https://github.com/sandraor/website/pull/3).

This one has a new iteration of my rake task for performing the geocoding of records in meetups.yml which are missing lat/lng values.

I also added a fix to the geolocation code for the case where the user does not permit the geolocation API to fetch the current location. Now, the map will display all the markers in this case.

**Comment by wifelette** [comment]
Summarizing in all possible places since at least _I_ can lose track of all the threads :p 
* There are two currently outstanding PRs:
 * https://github.com/sandraor/website/pull/10 by @solsberg needs some code removed so it doesn't conflict with the second open PR, and @wycats has provided some feedback on small other bits of the code. @solsberg says he'll work on this tonight.
 * https://github.com/sandraor/website/pull/9 by @alexclarkofficial is fixing a couple indentation changes, and then this is good to be merged.

Both folks should comment here to confirm when they're done with their final changes and ready to be merged in.

**Comment by solsberg** [comment]
I have updated https://github.com/sandraor/website/pull/10 to avoid the conflict with https://github.com/sandraor/website/pull/9
I also moved the main geocoding functionality from the Rakefile into a separate MeetupsData module - perhaps future code that has been discussed to make use of the metopes,

**Comment by wifelette** [comment]
There seems to be a slight bug, and it _might_ be one I created :p 

I pulled the most recent version to test, and it looks like the pin dropped on the map to represent *me*, the user, adds to the count. So in the Pacific NW, there are 4 meetups, but the zoomed out bubble shows 5, because it's 4 + the pin for me. Wondering if there's a way to have it not count the user's location. Maybe we don't need to show a pin for that at all? 

Also, if we _do_ need to, can we make it a different color? Right now it's a reddish that's similar to the Meetup pin, but just with a black dot on it when you zoom in.

None of these are deal breakers, and I haven't investigated them myself yet. Mentioning here in case anyone else has ideas and/or can get to it first.

REALLY excited about the open PR (https://github.com/sandraor/website/pull/11) that interfaces with Meetup! Can't wait to have it all come together :smile:

**Comment by sandraor** [comment]
@wifelette I see what you mean. It would be fine to hide the the user's location pin, or, if the pin can be customized, I can create a new pin image with a different colour. Regardless, it shouldn't be included in the pin cluster count.

There's another small task that needs to be done, if anyone is interested: If a pin is selected in the map, its corresponding list item needs to have an "active" class. I took a stab at it here, but it only works when you click a location in the list and does not change if you click another pin on the map: https://github.com/sandraor/website/commit/08eb91935ab8a60cc09768b939c0da95d06a32c4

**Comment by lzcabrera** [comment]
I took a look at the user's pin issue and just pushed a fix to not display it in sandraor#11 .

I'm really excited to see this live too :+1:

**Comment by lzcabrera** [comment]
@sandraor added the active class to the list item when pin is selected as part of sandraor#11 :)

**Comment by wifelette** [comment]
Current status, FYI: 
@wycats added a bunch of comments/suggestions to https://github.com/sandraor/website/pull/11. Once that's all sorted, we should be good to merge back in :)

@lzcabrera Poke me or @wycats if you have questions about his questions :p I'll make sure he responds.

**Comment by lzcabrera** [comment]
@wifelette i've incorporated @wycats suggestions in sandraor#11.

I think it is ready to be merged back in :)

**Comment by wifelette** [comment]
Awesome!

@sandraor I'll try and get @wycats and/or @rwjblue to do a final pass on the final product and give us a thumbs up.

ETA on the Resources page we're working on? I'm around via chat and Trello to sort out any details on that :) 

TL;DR: this is good to go/be merged (pending review) but we'll need to wait until we have the resources page + affiliated YAML file data to merge this. Hopefully it'll be pretty quick, and I don't anticipate any other changes to the page in the meantime, so we should stay current/able to be automatically merged.

**Comment by sandraor** [comment]
Is it possible to change the zoom levels of the map for different situations? When location detection is on, it's at zoom level 8, but centred on the user's location. However, when location detection is turned off, the map is centred in the middle of the ocean at the same zoom level, so it looks entirely blue. If it's possible to specify, these are the ideal zoom levels: 

Default zoom level: 2
Location detection enabled: 8
Pin selected: 14

**Comment by lzcabrera** [comment]
@sandraor I'm looking into the zoom levels and have a fix almost in place, I will be adding it to sandraor#12
While working on this fix I noticed the cluster images are returning a 404:

/images/meetups/map-cluster-1.png
If this is an an asset from a gem, add sprockets.import_asset 'map-cluster-1.png' to your config.rb.

I'm not sure what's going on here. The cluster images were working a few days ago and the images are in the right place.

**Comment by lzcabrera** [comment]
@sandraor checkout sandraor#13, I added the zoom levels fix

Anyone know how to fix the 404 errors on /images/meetups/map-cluster-1.png and /images/meetups/map-cluster-2.png?

When typing the path to those images on the browser, ie: http://localhost:4567/images/meetups/map-cluster-1.png I get:
"If this is an an asset from a gem, add sprockets.import_asset 'map-cluster-1.png' to your config.rb."

**Comment by yaymukund** [comment]
@lzcabrera I checked out `lzcabrera/meetup-list`, ran `bundle && bundle exec middleman` and visited http://0.0.0.0:4567/meetups and [it fetches both files without issue](https://cloud.githubusercontent.com/assets/590450/5791252/0dbb4f40-9e7d-11e4-8858-e90f4c55f4f5.jpg).

**Comment by lzcabrera** [comment]
Thanks for checking @yaymukund. I was still seeing the issues after running `bundle && bundle exec middleman`. I restarted my terminal and now it is working fine :)

**Comment by wifelette** [comment]
Just merged this back in and it's live! Woot!

Big thanks and super congrats to everyone who was learning the ropes on this one, especially @sandraor. Y'all did a great job, and I hope you'll stick around and help with future projects too <3

**Comment by owilliams** [comment]
Just wondering if anyone is working on the resources page? Is there a design or plan for it?

**Comment by wifelette** [comment]
Blocked on me, sad face :( Too much EmberConf!

@sandraor has ideas for the design and I'm needing to get her the content plan. Will be getting back to it as soon as I get everything settled again!

**Comment by owilliams** [comment]
pfft...excuses! :stuck_out_tongue_winking_eye: 

No, totally understandable! Let me know if I can help. And great job with EmberConf, loved it.
